broker/lib/middleware.js

```diff
@@ -44,6 +46,38 @@ exports.validateCreateRequest = function () {
   };
 };
 
+exports.validateSchemaForRequest = function (target, operation) {
+  return function (req, res, next) {
+    const plan = getPlanFromRequest(req);
+
+    const schema = _.get(plan, `schemas.${target}.${operation}.parameters`);
+
```

Comment:
Along with this, could you please also add few unit tests covering this middleware method?

Reply:
Yes, I have added this as TODO :)

Reply:
I have added tests